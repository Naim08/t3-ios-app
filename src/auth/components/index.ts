export { AuthInput } from './AuthInput';
export { ModalHeader } from './ModalHeader';
export { Divider } from './Divider';
export { BaseAuthModal } from './BaseAuthModal';