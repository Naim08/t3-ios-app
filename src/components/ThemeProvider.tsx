import React, {
  createContext,
  useContext,
  useState,
  useEffect,
  ReactNode,
} from 'react';
import { Appearance } from 'react-native';
import { Theme, createTheme, ColorScheme } from '../theme';

interface ThemeContextType {
  theme: Theme;
  colorScheme: ColorScheme;
  toggleTheme: () => void;
  setColorScheme: (scheme: ColorScheme) => void;
}

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

interface ThemeProviderProps {
  children: ReactNode;
}

export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
  const [colorScheme, setColorScheme] = useState<ColorScheme>(
    (Appearance.getColorScheme() as ColorScheme) || 'light',
  );

  useEffect(() => {
    const subscription = Appearance.addChangeListener(({ colorScheme }) => {
      setColorScheme((colorScheme as ColorScheme) || 'light');
    });

    return () => subscription.remove();
  }, []);

  const theme = createTheme(colorScheme);

  const toggleTheme = () => {
    setColorScheme((prev) => (prev === 'light' ? 'dark' : 'light'));
  };

  return (
    <ThemeContext.Provider
      value={{ theme, colorScheme, toggleTheme, setColorScheme }}
    >
      {children}
    </ThemeContext.Provider>
  );
};

export const useTheme = () => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};