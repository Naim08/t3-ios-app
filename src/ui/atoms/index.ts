export { Typography } from './Typography';
export type { TypographyProps } from './Typography';

export { PrimaryButton } from './PrimaryButton';
export type { PrimaryButtonProps } from './PrimaryButton';

export { TextField } from './TextField';
export type { TextFieldProps } from './TextField';

export { Surface } from './Surface';
export type { SurfaceProps } from './Surface';
