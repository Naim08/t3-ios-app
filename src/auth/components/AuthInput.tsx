import React from 'react';
import { View, TextInput, TextInputProps } from 'react-native';
import { Mail, Lock, User } from 'lucide-react-native';
import { useTheme } from '../../components/ThemeProvider';
import { authModalStyles } from '../styles/authModalStyles';

interface AuthInputProps extends Omit<TextInputProps, 'style'> {
  icon: 'mail' | 'lock' | 'user';
  hasError?: boolean;
}

const getIconComponent = (icon: AuthInputProps['icon'], theme: any): React.ReactElement => {
  const iconProps = { 
    size: 20, 
    color: theme.colors.textSecondary, 
    strokeWidth: 2 
  };
  
  switch (icon) {
    case 'mail':
      return <Mail {...iconProps} />;
    case 'lock':
      return <Lock {...iconProps} />;
    case 'user':
      return <User {...iconProps} />;
    default:
      return <Mail {...iconProps} />;
  }
};

export const AuthInput: React.FC<AuthInputProps> = ({
  icon,
  hasError = false,
  ...textInputProps
}) => {
  const { theme } = useTheme();

  return (
    <View 
      style={[
        authModalStyles.inputWrapper, 
        { 
          borderColor: hasError 
            ? 'rgba(255, 107, 107, 0.6)' 
            : textInputProps.value 
              ? theme.colors.brand['400'] 
              : 'rgba(0, 0, 0, 0.1)' 
        }
      ]}
    >
      <View style={authModalStyles.inputIcon}>
        {getIconComponent(icon, theme)}
      </View>
      <TextInput
        style={[authModalStyles.input, { color: theme.colors.textPrimary }]}
        placeholderTextColor={theme.colors.textSecondary}
        autoCapitalize="none"
        autoCorrect={false}
        {...textInputProps}
      />
    </View>
  );
};