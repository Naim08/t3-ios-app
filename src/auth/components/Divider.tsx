import React from 'react';
import { View, Text } from 'react-native';
import { useTheme } from '../../components/ThemeProvider';
import { authModalStyles } from '../styles/authModalStyles';

interface DividerProps {
  text?: string;
}

export const Divider: React.FC<DividerProps> = ({ text = 'OR' }) => {
  const { theme } = useTheme();

  return (
    <View style={authModalStyles.dividerContainer}>
      <View style={[authModalStyles.divider, { backgroundColor: theme.colors.border }]} />
      <Text style={[authModalStyles.dividerText, { color: theme.colors.textSecondary }]}>
        {text}
      </Text>
      <View style={[authModalStyles.divider, { backgroundColor: theme.colors.border }]} />
    </View>
  );
};