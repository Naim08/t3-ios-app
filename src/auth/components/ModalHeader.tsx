import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { X } from 'lucide-react-native';
import { useTheme } from '../../components/ThemeProvider';
import { authModalStyles } from '../styles/authModalStyles';

// Conditional imports for gradients
let LinearGradient: any, BlurView: any;
try {
  const gradientLib = require('expo-linear-gradient');
  LinearGradient = gradientLib.LinearGradient;
  const blurLib = require('expo-blur');
  BlurView = blurLib.BlurView;
} catch (error) {
  LinearGradient = ({ children, style, ...props }: any) => React.createElement(View, { style, ...props }, children);
  BlurView = ({ children, style, ...props }: any) => React.createElement(View, { style, ...props }, children);
}

interface ModalHeaderProps {
  title: string;
  onClose: () => void;
  showDragIndicator?: boolean;
}

export const ModalHeader: React.FC<ModalHeaderProps> = ({
  title,
  onClose,
  showDragIndicator = true,
}) => {
  const { theme } = useTheme();

  return (
    <LinearGradient
      colors={theme.isDark ? [
        '#1a1a2e',
        '#16213e',
        '#0f3460',
      ] : [
        '#ffffff',
        '#f8f9ff',
        '#f0f4ff',
      ]}
      style={authModalStyles.headerGradient}
      start={{ x: 0, y: 0 }}
      end={{ x: 1, y: 1 }}
    >
      {showDragIndicator && (
        <View style={authModalStyles.dragIndicator} />
      )}
      
      <View style={authModalStyles.header}>
        <Text style={[authModalStyles.title, { color: theme.colors.textPrimary }]}>
          {title}
        </Text>
        
        <TouchableOpacity 
          style={authModalStyles.closeButton}
          onPress={onClose}
          activeOpacity={0.7}
        >
          <BlurView
            style={authModalStyles.closeButtonBlur}
            intensity={20}
            tint={theme.isDark ? 'dark' : 'light'}
          >
            <X 
              size={18} 
              color={theme.colors.textSecondary} 
              strokeWidth={2.5} 
            />
          </BlurView>
        </TouchableOpacity>
      </View>
    </LinearGradient>
  );
};