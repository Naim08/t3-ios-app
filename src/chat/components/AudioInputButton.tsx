import React, { useRef, useEffect } from 'react';
import {
  View,
  TouchableOpacity,
  StyleSheet,
  Animated,
  Platform,
  ViewStyle,
} from 'react-native';
import { Mic, MicOff, X } from 'lucide-react-native';
import { useTheme } from '../../components/ThemeProvider';
import { Typography } from '../../ui/atoms';
import { useAudioRecording } from '../../hooks/useAudioRecording';

// Conditional import for gradient
let LinearGradient: any;
try {
  const gradientLib = require('expo-linear-gradient');
  LinearGradient = gradientLib.LinearGradient;
} catch (error) {
  LinearGradient = ({ children, style, ...props }: any) => 
    React.createElement(View, { style, ...props }, children);
}

interface AudioInputButtonProps {
  onRecordingComplete: (uri: string, duration: number) => void;
  onError?: (error: Error) => void;
  disabled?: boolean;
  style?: ViewStyle;
}

export const AudioInputButton: React.FC<AudioInputButtonProps> = ({
  onRecordingComplete,
  onError,
  disabled = false,
  style,
}) => {
  const { theme } = useTheme();
  const scaleAnim = useRef(new Animated.Value(1)).current;
  const pulseAnim = useRef(new Animated.Value(0)).current;
  const glowAnim = useRef(new Animated.Value(0)).current;

  const {
    isRecording,
    duration,
    startRecording,
    stopRecording,
    cancelRecording,
    hasPermission,
    requestPermission,
  } = useAudioRecording({
    onRecordingComplete,
    onError,
  });

  // Pulse animation when recording
  useEffect(() => {
    if (isRecording) {
      Animated.loop(
        Animated.sequence([
          Animated.timing(pulseAnim, {
            toValue: 1,
            duration: 1000,
            useNativeDriver: true,
          }),
          Animated.timing(pulseAnim, {
            toValue: 0,
            duration: 1000,
            useNativeDriver: true,
          }),
        ])
      ).start();

      // Glow animation
      Animated.timing(glowAnim, {
        toValue: 1,
        duration: 300,
        useNativeDriver: true,
      }).start();
    } else {
      pulseAnim.stopAnimation();
      Animated.timing(glowAnim, {
        toValue: 0,
        duration: 300,
        useNativeDriver: true,
      }).start();
    }
  }, [isRecording, pulseAnim, glowAnim]);

  const handlePressIn = () => {
    if (disabled) return;
    
    Animated.spring(scaleAnim, {
      toValue: 0.95,
      useNativeDriver: true,
      speed: 50,
      bounciness: 0,
    }).start();
  };

  const handlePressOut = () => {
    Animated.spring(scaleAnim, {
      toValue: 1,
      useNativeDriver: true,
      speed: 20,
      bounciness: 5,
    }).start();
  };

  const handlePress = async () => {
    if (disabled) return;

    if (isRecording) {
      await stopRecording();
    } else {
      if (!hasPermission) {
        const granted = await requestPermission();
        if (!granted) {
          onError?.(new Error('Microphone permission denied'));
          return;
        }
      }
      await startRecording();
    }
  };

  const handleCancel = async () => {
    await cancelRecording();
  };

  const formatDuration = (seconds: number) => {
    const mins = Math.floor(seconds / 60);
    const secs = seconds % 60;
    return `${mins}:${secs.toString().padStart(2, '0')}`;
  };

  const pulseScale = pulseAnim.interpolate({
    inputRange: [0, 1],
    outputRange: [1, 1.3],
  });

  const pulseOpacity = pulseAnim.interpolate({
    inputRange: [0, 1],
    outputRange: [0.3, 0],
  });

  const glowOpacity = glowAnim.interpolate({
    inputRange: [0, 1],
    outputRange: [0, 0.6],
  });

  return (
    <View style={[styles.container, style]}>
      {isRecording && (
        <>
          {/* Pulse effect */}
          <Animated.View
            style={[
              styles.pulse,
              {
                transform: [{ scale: pulseScale }],
                opacity: pulseOpacity,
                backgroundColor: theme.colors.danger['500'],
              },
            ]}
          />
          {/* Glow effect */}
          <Animated.View
            style={[
              styles.glow,
              {
                opacity: glowOpacity,
                shadowColor: theme.colors.danger['500'],
              },
            ]}
          />
        </>
      )}

      <Animated.View
        style={{
          transform: [{ scale: scaleAnim }],
        }}
      >
        <TouchableOpacity
          onPress={handlePress}
          onPressIn={handlePressIn}
          onPressOut={handlePressOut}
          disabled={disabled}
          activeOpacity={0.8}
          style={[
            styles.button,
            {
              backgroundColor: isRecording 
                ? theme.colors.danger['500']
                : disabled 
                  ? theme.colors.gray['300']
                  : theme.colors.brand['500'],
              shadowColor: isRecording 
                ? theme.colors.danger['500']
                : theme.colors.brand['500'],
            },
            disabled && styles.disabledButton,
          ]}
        >
          {isRecording ? (
            <LinearGradient
              colors={[
                theme.colors.danger['600'],
                theme.colors.danger['500'],
                theme.colors.danger['400'],
              ]}
              start={{ x: 0, y: 0 }}
              end={{ x: 1, y: 1 }}
              style={styles.gradientBackground}
            >
              <Mic size={24} color="#FFFFFF" strokeWidth={2.5} />
            </LinearGradient>
          ) : (
            <Mic 
              size={24} 
              color={disabled ? theme.colors.gray['500'] : '#FFFFFF'} 
              strokeWidth={2.5} 
            />
          )}
        </TouchableOpacity>
      </Animated.View>

      {/* Recording indicator */}
      {isRecording && (
        <View style={styles.recordingInfo}>
          <View style={styles.recordingDot} />
          <Typography 
            variant="caption" 
            weight="medium"
            style={{ color: theme.colors.danger['600'] }}
          >
            {formatDuration(duration)}
          </Typography>
          <TouchableOpacity
            onPress={handleCancel}
            style={styles.cancelButton}
            hitSlop={{ top: 10, bottom: 10, left: 10, right: 10 }}
          >
            <X size={16} color={theme.colors.gray['600']} strokeWidth={2.5} />
          </TouchableOpacity>
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    position: 'relative',
  },
  button: {
    width: 48,
    height: 48,
    borderRadius: 24,
    justifyContent: 'center',
    alignItems: 'center',
    ...Platform.select({
      ios: {
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.25,
        shadowRadius: 4,
      },
      android: {
        elevation: 6,
      },
    }),
  },
  disabledButton: {
    opacity: 0.6,
  },
  gradientBackground: {
    width: '100%',
    height: '100%',
    borderRadius: 24,
    justifyContent: 'center',
    alignItems: 'center',
  },
  pulse: {
    position: 'absolute',
    width: 48,
    height: 48,
    borderRadius: 24,
  },
  glow: {
    position: 'absolute',
    width: 56,
    height: 56,
    borderRadius: 28,
    ...Platform.select({
      ios: {
        shadowOffset: { width: 0, height: 0 },
        shadowOpacity: 1,
        shadowRadius: 20,
      },
      android: {
        elevation: 20,
      },
    }),
  },
  recordingInfo: {
    position: 'absolute',
    top: -30,
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: 'white',
    paddingHorizontal: 12,
    paddingVertical: 6,
    borderRadius: 16,
    ...Platform.select({
      ios: {
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 1 },
        shadowOpacity: 0.1,
        shadowRadius: 2,
      },
      android: {
        elevation: 2,
      },
    }),
  },
  recordingDot: {
    width: 6,
    height: 6,
    borderRadius: 3,
    backgroundColor: '#EF4444',
    marginRight: 6,
  },
  cancelButton: {
    marginLeft: 8,
  },
});