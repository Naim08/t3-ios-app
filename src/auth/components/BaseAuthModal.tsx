import React, { ReactNode } from 'react';
import {
  Modal,
  View,
  TouchableWithoutFeedback,
  KeyboardAvoidingView,
  Platform,
  ScrollView,
  Animated,
} from 'react-native';
import { useTheme } from '../../components/ThemeProvider';
import { useModalAnimation } from '../../hooks/useModalAnimation';
import { ModalHeader } from './ModalHeader';
import { authModalStyles } from '../styles/authModalStyles';

// Conditional import for BlurView
let BlurView: any;
try {
  const blurLib = require('expo-blur');
  BlurView = blurLib.BlurView;
} catch (error) {
  BlurView = View;
}

interface BaseAuthModalProps {
  visible: boolean;
  onClose: () => void;
  title: string;
  children: ReactNode;
  showDragIndicator?: boolean;
}

export const BaseAuthModal: React.FC<BaseAuthModalProps> = ({
  visible,
  onClose,
  title,
  children,
  showDragIndicator = true,
}) => {
  const { theme } = useTheme();
  const { slideAnim, fadeAnim, scaleAnim } = useModalAnimation({ visible });

  const handleBackdropPress = () => {
    onClose();
  };

  return (
    <Modal
      visible={visible}
      transparent
      animationType="none"
      statusBarTranslucent
      onRequestClose={onClose}
    >
      <KeyboardAvoidingView
        style={{ flex: 1 }}
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
      >
        <View style={authModalStyles.overlay}>
          {/* Animated Backdrop */}
          <Animated.View
            style={[
              authModalStyles.backdrop,
              { opacity: fadeAnim }
            ]}
          >
            <TouchableWithoutFeedback onPress={handleBackdropPress}>
              {Platform.OS === 'ios' ? (
                <BlurView
                  intensity={80}
                  tint="dark"
                  style={authModalStyles.backdropTouch}
                />
              ) : (
                <View style={authModalStyles.backdropTouch} />
              )}
            </TouchableWithoutFeedback>
          </Animated.View>

          {/* Animated Modal Container */}
          <Animated.View
            style={[
              authModalStyles.container,
              {
                backgroundColor: theme.colors.surface,
                transform: [
                  { translateY: slideAnim },
                  { scale: scaleAnim }
                ],
              },
            ]}
          >
            <ScrollView
              style={{ flex: 1 }}
              contentContainerStyle={authModalStyles.scrollContent}
              keyboardShouldPersistTaps="handled"
              showsVerticalScrollIndicator={false}
            >
              <ModalHeader
                title={title}
                onClose={onClose}
                showDragIndicator={showDragIndicator}
              />
              
              {children}
            </ScrollView>
          </Animated.View>
        </View>
      </KeyboardAvoidingView>
    </Modal>
  );
};