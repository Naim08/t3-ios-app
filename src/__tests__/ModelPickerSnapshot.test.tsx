import React from 'react';
import { View } from 'react-native';
import { render } from '@testing-library/react-native';
import { Typography } from '../ui/atoms';
import { ThemeProvider } from '../components/ThemeProvider';

// Import jest globals
import { describe, test, expect } from '@jest/globals';

describe('ModelPicker', () => {
  test('ModelPickerSnapshot', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <View>
          <Typography variant="h3">Model Picker Test</Typography>
        </View>
      </ThemeProvider>
    );
    
    expect(toJSON()).toMatchSnapshot();
  });
});
