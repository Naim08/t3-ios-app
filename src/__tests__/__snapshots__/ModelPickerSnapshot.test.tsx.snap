// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelPicker ModelPickerSnapshot 1`] = `
<View>
  <Text
    accessibilityRole="text"
    allowFontScaling={true}
    maxFontSizeMultiplier={3}
    style={
      [
        {
          "includeFontPadding": false,
        },
        {
          "color": "#212529",
          "fontFamily": "System",
          "fontSize": 24,
          "fontWeight": "400",
          "lineHeight": 32,
          "textAlign": "left",
        },
      ]
    }
  >
    Model Picker Test
  </Text>
</View>
`;
